src/utils/gitHubUtils.ts

```diff
@@ -104,12 +104,14 @@ export async function getGitHubQuickPicksWithLoadMore<T>(cache: ICachedQuickPick
     }
 }
 
-export async function createGitHubRequestOptions(context: IStaticWebAppWizardContext, url: string, method: HttpMethods = 'GET'): Promise<gitHubWebResource> {
-    if (!context.accessToken) {
-        context.accessToken = await getGitHubAccessToken();
+export async function createGitHubRequestOptions(context: IGitHubContext | undefined, url: string, method: HttpMethods = 'GET'): Promise<gitHubWebResource> {
+    const accessToken: string = context?.accessToken || await getGitHubAccessToken();
+
+    if (context && !context.accessToken) {
+        context.accessToken = accessToken;
```

Comment:
The context is optional and is used when creating StaticWebApps.  I just set it so that way I don't have to keep retrieving it from the VS Code API, but honestly that's probably a safer and quicker way to do it.

Reply:
Ah right, the one annoying thing about this though is that every time we make a GitHub request, we have to get it from VS Code.  That's fine if you set it to never ask me for permissions again, but if a user didn't enable that, it'd be super annoying (and we shouldn't have to ask the user for a token 3 times in one static web app create)

Reply:
I'm not opposed to setting accessToken on context - I'm opposed to _where_ you're doing it. I don't like functions that have unrelated side-effects. You should be able to add accessToken to context when you're setting up the wizard (pretty standard procedure to add a few default values to a wizardContext)

Btw, there is a bit of risk the accessToken expires during the course of a single command, but I think that's outweighed by the extra prompts you mentioned.

Reply:
Actually, it seems like VS Code may have gotten rid of the prompts for GitHub access (unless I can't figure how to turn off the 'Don't Show me Again" setting).  Still, I imagine VS Code's Token API is making a call to GitHub which could potentially increase our latency between steps which seems annoying.